src/icrar/leap-accelerate/algorithm/casa/PhaseRotate.cc

```diff
@@ -92,21 +88,27 @@ namespace casalib
                 metadata.GetBaselines(),
                 metadata.num_pols));
 
-            assert(48 == queue.front().data.dimension(0)); //metadata.channels
-            assert(8001 == queue.front().data.dimension(1)); //metadata.baselines
-            assert(4 == queue.front().data.dimension(2)); //metadata.polarizations
+#if _DEBUG
+            assert(metadata.GetChannels() == queue.front().data.dimension(0)); //metadata.channels
+            assert(metadata.GetBaselines() == queue.front().data.dimension(1)); //metadata.baselines
+            assert(metadata.GetPolarizations() == queue.front().data.dimension(2)); //metadata.polarizations
             std::cout << "==== integration data " << queue.front().data.dimensions() << std::endl;
             std::cout << "==== integration data(0,0)" << queue.front().data.chip(0, 0).chip(0, 0) << std::endl;
             std::cout << "==== integration data(1,0)" << queue.front().data.chip(1, 0).chip(0, 0) << std::endl;
             std::cout << "==== integration data(2,0)" << queue.front().data.chip(2, 0).chip(0, 0) << std::endl;
             std::cout << "==== integration data(3,0)" << queue.front().data.chip(3, 0).chip(0, 0) << std::endl;
             std::cout << "==== integration data(4,0)" << queue.front().data.chip(4, 0).chip(0, 0) << std::endl;
-
+#endif
             input_queues.push_back(queue);
             output_integrations.push_back(std::queue<IntegrationResult>());
             output_calibrations.push_back(std::queue<CalibrationResult>());
         }
 
+#if _DEBUG
+        std::cout << "direction count " << directions.size() << std::endl;
+        std::cout << "input count " << input_queues.size() << std::endl;
+#endif
+
```

Comment:
Same as above, prefer using the standard `NDEBUG` macro, or simply remove the printing to stdout